src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseRepositoryService.java

```diff
@@ -229,6 +229,13 @@ void createRepositoriesForNewExercise(final ProgrammingExercise programmingExerc
         createAndInitializeAuxiliaryRepositories(projectKey, programmingExercise);
     }
 
+    /**
+     * Creates and initializes all auxiliary repositories for a new programming exercise.
+     *
+     * @param projectKey          The key of the project the exercise belongs to. Can be found in the programming exercise.
+     * @param programmingExercise The programming exercise for which the auxiliary repositories should be created.
+     * @throws GitAPIException Thrown in case creating a repository fails.
+     */
     private void createAndInitializeAuxiliaryRepositories(final String projectKey, final ProgrammingExercise programmingExercise) throws GitAPIException {
```

Comment:
For a follow up:
If we already pass the programming exercise that contains the project key, why do we need to pass the key separately as well?

Reply:
I think that the auxiliary repositories are generated before the exercise is saved (if I remember correctly)